docs/source/working_with_aiida/caching.rst

```diff
@@ -4,45 +4,90 @@
 Caching
 *******
 
-When working with AiiDA, you might sometimes re-run calculations which were already successfully executed. Because this can waste a lot of computational resources, you can enable AiiDA to **cache** calculations, which means that it will re-use existing calculations if a calculation with the same inputs is submitted again.
+Enabling caching
+----------------
 
-When a calculation is cached, a copy of the original calculation is created. This copy will keep the input links of the new calculation. The outputs of the original calculation are also copied, and linked to the new calculation. This allows for the new calculation to be a separate Node in the provenance graph and, critically, preserves the acyclicity of the graph.
+There are numerous reasons why you may need to re-run calculations you’ve already done before.
+Since AiiDA stores the full provenance of each calculation, it can *detect* whether a calculation has been run before
+and reuse its outputs without wasting computational resources.
 
-Caching is also implemented for Data nodes. This is not very useful in practice (yet), but is an easy way to show how the caching mechanism works:
+This feature is not (yet) enabled by default. In order to enable hashing:
```

Comment:
this is not enabling "hashing" but it is enabling "caching". We need to define and explain these terms better. We are using the terms "hashing" and "caching" too loosely and interchangeably. The "caching" only makes sense for `ProcessNodes`, where, for reasons explained later on, this is disabled for `WorkflowNodes` and therefore only possible for `CalculationNodes`. The "hashing" is merely a method to detect equivalent nodes and therefore allow to implement the "caching".

Reply:
you are right and I will adapt.